backend/services/services.py

```
@@ -0,0 +1,213 @@
+        raise HTTPException(status_code=404, detail="Event details not found")
+    
+    return post, event
+
+# Helper function to convert event date and time to UTC
+def convert_to_utc(event_date: str, event_time: str, user_timezone: str) -> Optional[datetime]:
+    try:
+        local_datetime = datetime.strptime(f"{event_date} {event_time}", "%Y-%m-%d %H:%M")
+        local_tz = ZoneInfo(user_timezone)  # Correct way to use ZoneInfo
+        local_dt_with_tz = local_datetime.replace(tzinfo=local_tz)  # Add timezone info
+        return local_dt_with_tz.astimezone(ZoneInfo("UTC"))  # Convert to UTC
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)